Add breadcrumbs to make it easier to tell the Chime name while in folder

jxjj opened the pull request:
- Adds a breadcrumb component
- Adds breadcrumbs to ChimePage and FolderPage

Breadcrumb component:
<img width="500" alt="Screen Shot 2022-09-15 at 5 30 01 PM" src="https://user-images.githubusercontent.com/980170/190519872-b8eff1ef-55bf-469d-8ab1-a4220873a744.png">

Folder Page with Breadcrumb:
<img width="500" alt="Screen Shot 2022-09-15 at 5 28 05 PM" src="https://user-images.githubusercontent.com/980170/190519896-b2642296-dd9f-42ce-b51f-1fe0b8aaa00d.png">

Design notes:
- Tried it both with and without current page showing as the last item. I thought it was a little better with the current page. Even though there's duplication of text, I thought it added clarity about the Home > Chime > Folder> Question organization that could aid new users.
- Using a basic "/" as a divider. Tried a chevron, but it looked a little strange with the "back" chevron pointing in the opposite direction.

Definitely weakly held opinions though, so let me know what you think.

Fixes #466

cmcfadden commented:
We should play around with some more real course titles and folder combos - departments really love doing these absurdly long descriptive titles, which Canvas passes through to us. An example of one I just helped with was:
`HIST 1301W (001) Authority and Rebellion: American History to 1865 (Fall 2022)` / `Lecture Participation Questions`

Is there anything we could do in the nav bar region  ("back to chime") or potentially the "College of Liberal Arts" region, which might be more subtle but still useful? I'm wondering if this tradeoff is worth it (pushing controls a bit further down the screen, and with a long title like the one above adding visual noise). 

My sense of the problem we're trying to solve is mostly when a user launches from Canvas and isn't sure if they're looking at this semester's chime or last semester's chime - does that seem reasonable? (My assumption is that if you come in through the front door and have clicked into a Chime, you're pretty certain on where you are)

jxjj commented:
Thanks for looking at this with me.

> We should play around with some more real course titles and folder combos - departments really love doing these absurdly long descriptive titles, which Canvas passes through to us. An example of one I just helped with was:
`HIST 1301W (001) Authority and Rebellion: American History to 1865 (Fall 2022) / Lecture Participation Questions`

Agreed.

Yesterday, I tried using a max-width and then ellipsis for text-overflow, but it cut off the useful part of the course name.

I wonder if using a condensed font would help?

> Is there anything we could do in the nav bar region ("back to chime") or potentially the "College of Liberal Arts" region, which might be more subtle but still useful?

The navbar region is a possibility, but has some complexity. When I explored it – one issue I ran into was it collapsing on medium and small screens, so it hides away the course name and turns it into buttons competing with the Back button.

> My sense of the problem we're trying to solve is mostly when a user launches from Canvas and isn't sure if they're looking at this semester's chime or last semester's chime - does that seem reasonable? 

I think that's probably the largest use case. 

The other case  is instructors will often find a naming schema they like and stick with it across courses. For example, if I'm teaching teaching Calc 1 and Calc 2, I might name my folders: "Lecture 1.1", "Lecture 1.2" in both courses.

> (My assumption is that if you come in through the front door and have clicked into a Chime, you're pretty certain on where you are)

Here, I might disagree. 

I kinda think of it like good signage. Even though you might walk by a big sign that says "Concourse A" or "3rd Floor", it's still super helpful to see those concourse letters on the gate signs ("A14") or that the 3rd floor room numbers that begin with a 3– especially if you're unfamiliar with the place and moving quickly. It reduces cognitive load.

Clicking through Chimein has even tripped me up a few times – especially when I'm helping other faculty who might be teaching in the same discipline. I start editing and then worry that I'm in the wrong course, since their folder names are all a bit generic.


Anyway, I think getting the course name on the chime folder page is where the value is. We could ditch the breadcrumb idea, and just put the course name as a link?

Here's a mock just using the course name, no breadcrumbs, and using Open Sans Condensed for the typeface. Also, I moved the buttons to the side to give those titles a bit more room.

<img width="1210" alt="Screen Shot 2022-09-16 at 4 49 48 AM" src="https://user-images.githubusercontent.com/980170/190611877-569d8480-e777-4a66-b75a-c3f6649e8d66.png">

Smaller screen view:
<img width="417" alt="Screen Shot 2022-09-16 at 4 53 56 AM" src="https://user-images.githubusercontent.com/980170/190612535-77abdf52-0773-4c27-b51c-7012a91e5398.png">

Let me know which direction you think is most promising.

cmcfadden commented:
Doing just the course name above the folder feels like a really good fit to me. If it's something we want to roll in soon-ish, we should just do that change, without rearranging the button structure - changing that UI mid-semester would create some tension for our less confident users.

jxjj commented:
Updated to just a link on the Folder Page.

Screenshot:
<img width="1490" alt="Screen Shot 2022-09-16 at 2 39 25 PM" src="https://user-images.githubusercontent.com/980170/190720982-eb843dce-60c1-4474-8bae-b7428acaa8ac.png">

While there, I fixed a few other small issues too:
- got rid of borders on individual buttons, which resolves #151 and removes some visual noise.
- fixed a wrapping issue on folder page
- Made folder settings panel toggle work like the Chime settings panel toggle and added a background to the panel. 
- Changed the alert dismiss button from a letter "X" to a close icon.

Latest version on: https://cla-chimein-dev.cla.umn.edu/

cmcfadden commented:
Love it!!
